### PR TITLE
Add @NotBlank @NotEmpty  with message

### PR DESCRIPTION
**Pull request by arturobernalg**
Add @NotBlank @NotEmpty  with message  support. The annotation @NotBlank, @NotNull and @NotEmpty often come with information about the error.

basically the idea it's make able add custom message to the annotations.

 @NotNull(message = "Name may not be null")
@NotEmpty(message = "Name may not be empty")
@NotBlank(message = "Name may not be blank")

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by Zomzog** [comment]
Hi,
You need to run `./bin/generate-samples.sh` to generate all samples for fixing CircleCi build.
Don't forget to also copy the technical committee.

I don't understand why you required a vendor extension for this. For me, not-empty is a constrain with minLength / maxLength constraints. For not-blank, it's more tricky but maybe a pattern.

**Comment by arturobernalg** [comment]
executed -->  ./bin/generate-samples.sh. I have committed the modified files

**Comment by davide1995** [comment]
If I understand correctly the reason for this commit it will mean that when I will write a field as 'required' in my yaml file, @NotEmpty also appears in the generated java file together with the actual @NotNull, right?

File: modules/openapi-generator/src/main/resources/JavaSpring/beanValidation.mustache [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {{#required}}
-  @NotNull
+    {{^vendorExtensions.x-snt-validation-message-not-blank}} {{^vendorExtensions.x-snt-validation-message-not-empty}} @NotNull {{#vendorExtensions.x-snt-validation-message-not-null}}(message = "{{{vendorExtensions.x-snt-validation-message-not-null}}}"){{/vendorExtensions.x-snt-validation-message-not-null}}       {{/vendorExtensions.x-snt-validation-message-not-empty}} {{/vendorExtensions.x-snt-validation-message-not-blank}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaSpring/beanValidation.mustache:2]: @arturobernalg thanks for the PR. What does "snt" stand for?

Comment (arturobernalg) [reply modules/openapi-generator/src/main/resources/JavaSpring/beanValidation.mustache]: HI @wing328 ...it's a mistake.i will change it

Comment (arturobernalg) [reply modules/openapi-generator/src/main/resources/JavaSpring/beanValidation.mustache]: change it. thanks

**Comment by wing328** [comment]
@arturobernalg thanks again for the PR. May I suggest the following instead?

1. Use `vendorExtensions.x-java-annotations` if defined. Obtained it from the spec.
2. If `vendorExtensions.x-java-annotations` is not defined, use the existing logic/template in  `beanValidation.mustache`

This way users can include whatever annotations they want without the overhead to define a vendor extension per annotation.

What do you guys think?

**Comment by davide1995** [comment]
Any update on the merge and when will be available in a future version?

**Pull request closed without merging**